### ISSUE TITLE
API Call do not pass $action value ...

### ISSUE THREAD

**Issue by rycks** [Dolibarr/dolibarr#27202]
### Bug

Here is an example with "afterPDFCreation" hook, as you can see action= is empty !

```
HookManager::executeHooks Qualified hook found (hooktype=output). We call method afterPDFCreation of class ActionsFacturx, module=facturx, action= context=pdfgeneration
```

**Comment by atm-maxime** [comment]
Hi @rycks !
Can this come from your facturx pdf model ? In crabe module I have this :
![image](https://github.com/Dolibarr/dolibarr/assets/2097773/033faa28-19bf-4d8f-93e3-c319ce2a82da)

Seems you need to have global $action defined to have the action given to the hook.

**Comment by rycks** [comment]
hello @atm-maxime 
yes it is but when that hook is called from API request $action is empty ... maybe i could send a example with builddoc api request ?

**Comment by atm-maxime** [comment]
Maybe that's normal. From the builddoc API there's no action (unlike as a user when action is validate for example)

**Comment by rycks** [comment]
yes but as hook is called i think that's a bug or something like a bug

i think i could be "full called" (ie with action) or "not called at all" (no hook in api mode it could be "more" logic) ...
